### PR TITLE
Validate the input samplesheet against the schema matching --source

### LINKED ISSUE #258: Non-fastq --source values abort at startup: PIPELINE_INITIALISATION validates every samplesheet against the fastq schema
### Description of the bug

On `dev`, every non-fastq `--source` aborts before the workflow starts, because `PIPELINE_INITIALISATION` validates `params.input` against the **fastq** samplesheet schema unconditionally.

`subworkflows/local/utils_nfcore_rnasplice_pipeline/main.nf:110`:

```groovy
channel
    .fromList(samplesheetToList(input, "${projectDir}/assets/schema_input.json"))
    .map {
        meta, fastq_1, fastq_2 ->
        ...
    }
    .set { ch_samplesheet }
```

`assets/schema_input.json` requires `fastq_1` and `strandedness`, so a `genome_bam`, `transcriptome_bam` or `salmon_results` samplesheet fails validation and the run dies. The pipeline does have a dedicated schema for each source (`assets/schema_input_genome_bam.json`, `assets/schema_input_transcriptome_bam.json`, `assets/schema_input_salmon_results.json`) and `workflows/rnasplice.nf` picks the right one in its `if (params.source == ...)` chain, but the run never gets that far.

Two things make this a pure false negative:

1. The schema is hardcoded rather than selected from `params.source`.
2. The channel built here, `PIPELINE_INITIALISATION.out.samplesheet`, is **never consumed**. `main.nf` calls `PIPELINE_INITIALISATION(...)` and then `NFCORE_RNASPLICE()` with no arguments, and `NFCORE_RNASPLICE` re-reads `params.input` itself (`ch_samplesheet = channel.value(file(params.input, checkIfExists: true))`, `main.nf:214`), as does `RNASPLICE` for its per-source parsing.

So the only effect of this block on a non-fastq run is to abort it.

This affects `-profile test_genome_bam`, `-profile test_transcriptome_bam` and `-profile test_salmon_results`, i.e. three of the pipeline's advertised entry points. The three test samplesheets on the test-datasets `rnasplice` branch all lack `fastq_1` and `strandedness`, as they should.

The hardcoded `assets/schema_input.json` is nf-core template boilerplate that came in with the template migration on `dev` (it is present since the 3.0.0 template update and survived 4.0.3). `master` / v1.0.5 predates this subworkflow entirely — it used `subworkflows/local/input_check.nf` — so this looks like an unreleased-only regression.

Found while adding a pipeline level nf-test in #257; that test uses `--source fastq` to work around it.

### Command used and terminal output

```console
$ nextflow run . -profile test_genome_bam,docker --outdir out -stub

[WARN] WARN: Both --aligner=star_salmon and --pseudo_aligner=salmon specified. Downstream analyses will be performed on both salmon output files.
[ERROR] ERROR ~ Validation of samplesheet failed!

 -- Check '.nextflow.log' file for details

[FAILED] completed=0 failed=0 cached=0
The following errors have been detected in /nf-core/test-datasets/rnasplice/samplesheet/samplesheet_bam.csv:

-> Entry 1: Missing required field(s): strandedness, fastq_1
-> Entry 2: Missing required field(s): strandedness, fastq_1
-> Entry 3: Missing required field(s): strandedness, fastq_1
-> Entry 4: Missing required field(s): strandedness, fastq_1

 -- Check script 'subworkflows/local/utils_nfcore_rnasplice_pipeline/main.nf' at line: 110 or see '.nextflow.log' file for more details
```

Not a single process is submitted; the failure is at initialisation.

### Relevant files

- `subworkflows/local/utils_nfcore_rnasplice_pipeline/main.nf` (the `samplesheetToList` call and the unused `emit: samplesheet`)
- `main.nf` (`PIPELINE_INITIALISATION` call site, and `NFCORE_RNASPLICE` re-reading `params.input`)
- `assets/schema_input*.json`
- `conf/test_genome_bam.config`, `conf/test_transcriptome_bam.config`, `conf/test_salmon_results.config`

### System information

- Nextflow 26.04.1.12112
- Docker, local executor, WSL2 (Linux 6.6.87.2-microsoft-standard-WSL2)
- rnasplice `dev` @ 1.1.0dev

### Possible fix

Either select the schema from `params.source` in `PIPELINE_INITIALISATION` the same way `RNASPLICE` does, or drop the channel entirely, since nothing consumes it. The second is smaller and removes the duplicated samplesheet parsing, but the first keeps an early, actionable validation error for all four sources, which is what the block was presumably there for.

Whichever way it goes, it needs a pipeline level nf-test per source to stop it regressing again — none of the non-fastq entry points is covered by `nf-test` today.

### PR DESCRIPTION
**Pull request by piplus2**
## Summary

Closes #258. `PIPELINE_INITIALISATION` validated every samplesheet against `assets/schema_input.json`, the fastq schema, so `--source genome_bam`, `--source transcriptome_bam` and `--source salmon_results` all aborted at startup with `Missing required field(s): strandedness, fastq_1`, before a single process was submitted. Three of the four advertised entry points were unusable on `dev`.

### Fixed

- **The schema is chosen from `--source`.** `samplesheetSchema()` is the one place that maps a source to its schema, and it errors on an unknown `--source` instead of silently validating against the wrong one.

- **The samplesheet is read exactly once.** `RNASPLICE` parses it into `samplesheet_rows` with the schema `samplesheetSchema(params.source)` selects, builds `ch_reads` from those rows, and derives the rMATS single condition check from the same rows. Previously the samplesheet was parsed twice by two independent code paths, in `PIPELINE_INITIALISATION` and again in `RNASPLICE`, which is what let the two disagree.

  Concretely that removes: the dead per-source parse in `PIPELINE_INITIALISATION`, whose emitted channel nothing consumed and whose only effect was to abort non-fastq runs; a second `channel.value(file(params.input))` built at the `RMATS` call site instead of the `ch_samplesheet` the workflow already receives; and the hand-rolled `new File(params.input).readLines()` / `new URL(...)` CSV parsing behind the rMATS single condition check, which only handled local and `http` paths and split on commas with no schema at all.

  The layout follows rnaseq: `PIPELINE_INITIALISATION` takes `_input` without using it and emits nothing, `NFCORE_RNASPLICE` takes no inputs and builds `ch_samplesheet` and `ch_contrastsheet` itself, and the workflow does the parsing. `main.nf` is unchanged from `dev`.

Fixing the abort uncovered three latent bugs in the paths it unblocked. Each one aborted the run, and none was reachable before, which is why they had gone unnoticed:

- `BAM_SORT_STATS_SAMTOOLS` was called with a bare fasta path in the `genome_bam` and `transcriptome_bam` paths. It declares `[ meta, fasta, fai ]`, so the run died with `Input tuple does not match tuple declaration in process ... SAMTOOLS_SORT -- offending value: X.fa`.
- The `salmon_results` path with `--suppa` referenced `ch_txi_suppa_tpm`, which is only assigned in the `star_salmon` branch: `No such variable: ch_txi_suppa_tpm`.
- `ISOFORMSWITCHANALYZER` received the `.tar.gz` archives rather than the Salmon directories, failing with `There were no directories containing the file names/suffixes typically generated by Kallisto/Salmon/RSEM/StringTie`. `TX2GENE_TXIMPORT` already extracts them with `UNTAR` for its own use and now emits them.

### Added

- `tests/genome_bam.nf.test`, `tests/transcriptome_bam.nf.test` and `tests/salmon_results.nf.test`, pipeline level tests for the three sources that had no coverage at all. They reuse the existing `test_genome_bam`, `test_transcriptome_bam` and `test_salmon_results` profiles and snapshot the same way `tests/default.nf.test` does.

## Testing

All four sources now run to completion:

```
nf-test test tests/default.nf.test           PASSED (359.95s)  # unchanged snapshot, no fastq regression
nf-test test tests/genome_bam.nf.test        PASSED (204.87s)
nf-test test tests/transcriptome_bam.nf.test PASSED ( 94.91s)
nf-test test tests/salmon_results.nf.test    PASSED ( 76.28s)
```

The fastq result is the important one for regression: `tests/default.nf.test` matches its existing snapshot, so the refactor is behaviour preserving for the only source that worked before.

Each new snapshot was recorded and then verified in a second run against the recorded file.

`prettier` and `nextflow lint` are clean on the changed files.

## Note for reviewers

`-stub` runs are not usable to check this end to end: `MERGEEVENTS` has a stub that writes `${prefix}.ioe`, which collides with a staged input of the same name, so Nextflow excludes it from the output matching set and the process fails with `Missing output file(s) '*.ioe'`. That is independent of this PR and affects every source, including fastq. All the verification above is from real runs. Worth its own issue.

Also left alone, as it is the same pattern one level over: the contrastsheet is still validated in `RNASPLICE` by building a `ch_contrasts` channel that nothing consumes, while `RMATS` and `SUPPA` re-read the raw CSV with `splitCsv`. Giving the contrastsheet the same treatment as the samplesheet would be a natural follow up.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01AcWhAN6Nmpwd9Juj5525bL

### CODE REVIEW

File: subworkflows/local/tx2gene_tximport/main.nf [hunk 1]
```diff
@@ -47,6 +48,7 @@ workflow TX2GENE_TXIMPORT {
     TXIMPORT(ch_salmon_results, tx2gene)
 
     emit:
+    salmon_results                  = ch_untarred_results // tuple [meta, salmon dir], extracted if the input was a tarball
```
Comment (jonasscheid) [inline subworkflows/local/tx2gene_tximport/main.nf]: Maybe this and then rm the unnecessary variable defintion

```suggestion
    salmon_results                  = salmon_results.dir.mix(UNTAR.out.untar) // tuple [meta, salmon dir], extracted if the input was a tarball
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -84,6 +85,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - #256 - Fixed the iGenomes STAR alignment path: `STAR_ALIGN_IGENOMES` pins STAR 2.6.1d, which does not accept `--quantTranscriptomeSAMoutput`, so it now uses the equivalent `--quantTranscriptomeBan Singleend` (by @piplus2)
 - #257 - Fixed `--gff_dexseq` aborting the pipeline with `Not a valid argument for 'combine' operator [String]`. `RNASPLICE` overwrote the `ch_dexseq_gff` it already receives from `PREPARE_GENOME` with the raw `params.gff_dexseq` string; that assignment is now removed, so the channel prepared by `PREPARE_GENOME` (including the `GUNZIP_GFF_DEXSEQ` step for a `.gff.gz` input) reaches `DEXSEQ_DEU` (by @piplus2)
+- #258 - Fixed `--source genome_bam`, `--source transcriptome_bam` and `--source salmon_results` aborting at startup. `PIPELINE_INITIALISATION` validated every samplesheet against `assets/schema_input.json`, the fastq schema, so a samplesheet for any other source failed with `Missing required field(s): strandedness, fastq_1` before a single process was submitted. The schema is now selected from `--source` by `samplesheetSchema()`, which is the one place that maps a source to its schema (by @piplus2)
```
Comment (jonasscheid) [inline CHANGELOG.md]: This is style, but I would say the changelog should not contain every commit and reasoning that was made (by AI). Schould be human-digestable imo

File: workflows/rnasplice.nf [hunk 3]
```diff
@@ -82,8 +83,16 @@ workflow RNASPLICE {
     //
     // Create channel from input file provided through params.input
     //
+    // The samplesheet columns depend on --source, so it is validated against the schema
```
Comment (jonasscheid) [inline workflows/rnasplice.nf]: This is also style but the samplesheet generation traditionally is handled in pipeline initialization (lines 86 to roughly 180). Any reason not to outsource this?

**Review by jonasscheid — CHANGES_REQUESTED** [review]
I like the idea of how you validate with mutliple input types! 2,3 open points from my side

File: subworkflows/local/tx2gene_tximport/main.nf (hunk 1 above)
Comment (piplus2) [reply subworkflows/local/tx2gene_tximport/main.nf]: thanks! removed the redundant variable

File: CHANGELOG.md (hunk 2 above)
Comment (piplus2) [reply CHANGELOG.md]: made it shorter, thanks!

File: workflows/rnasplice.nf (hunk 3 above)
Comment (piplus2) [reply workflows/rnasplice.nf]: There is, and it's the reason this PR exists. Because the columns depend on `--source`, the samplesheet is a per-source thing rather than a fixed one, and #258 came precisely from it being handled in two places: initialisation validated it against the fastq schema while the workflow parsed it again per source, so the three non-fastq sources died at startup. Keeping the parse in one place is what makes it impossible for the two to disagree again.

I did start with it in initialisation, but to hand the parsed channel over, `PIPELINE_INITIALISATION` has to emit it and the entry workflow has to feed it into `NFCORE_RNASPLICE`, which then no longer takes zero inputs like the template does. Between the two conventions I kept the entry workflow shape and moved the parse down, leaving initialisation to do what it's really for: validating the parameters, `--source` included, so a bad value still stops the run before anything is submitted.

Happy to swap it round if you'd rather have it in initialisation — it just costs `NFCORE_RNASPLICE` its empty input list.

Comment (jonasscheid) [reply workflows/rnasplice.nf]: Sorry, I don't understand the argument. You can put the samplesheet handling into pipeline initialization and emit all the necessary channels that are needed beyond line 180, no?

Comment (piplus2) [reply workflows/rnasplice.nf]: Yes but that means that NFCORE_RNASPLICE must get some inputs - the relevant output of PIPELINE_INITIALISATION - which is against the template.

Comment (jonasscheid) [reply workflows/rnasplice.nf]: 🤔 Sorry, no, the template explicitly takes input channels: https://github.com/nf-core/tools/blob/eb2f709090f4054f45437c34049ea2068567c339/nf_core/pipeline-template/workflows/pipeline.nf#L29

In any case, this is not blocking, but before release 1.0.0 I would move it to pipeline initialisation, opening an issue such that its documented :)

**Review by jonasscheid — APPROVED** [review]

**Comment by piplus2** [comment]
thanks @jonasscheid !
for the time beings I'm passing both the path and parsed samplesheet as they are both required. In the next PR I'm going to streamline the handling of the samplesheet and contrastsheet.
